pkg/deploy/deployment.go

```diff
@@ -111,10 +126,102 @@ func GetClusterDeployment(name string, namespace string, client runtimeClient.Cl
 	}
 	err := client.Get(context.TODO(), namespacedName, deployment)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			return nil, nil
 		}
 		return nil, err
 	}
 	return deployment, nil
 }
+
+func MountSecrets(specDeployment *appsv1.Deployment, deployContext *DeployContext) error {
```

Comment:
Please add description that explains what this function actually does (what secrets, why)